librosa/effects.py

```diff
@@ -660,3 +661,68 @@ def preemphasis(y, coef=0.97, zi=None, return_zf=False):
         return y_out, z_f
 
     return y_out
+
+
+def deemphasis(y, coef=0.95, zi=None):
+    """De-emphasize an audio signal with the inverse operation of preemphasis():
+
+    If y = preemphasis(x, coef=coef, zi=zi), the deemphasis is:
+        x[i] -> y[i] + coef * x[i-1]
+        x = deemphasis(y, coef=coef, zi=zi)
+
+
+    Parameters
+    ----------
+    y : np.ndarray
+        Audio signal
+
+    coef : positive number
+        Pre-emphasis coefficient.  Typical values of ``coef`` are between 0 and 1.
+
+        At the limit ``coef=0``, the signal is unchanged.
+
+        At ``coef=1``, the result is the first-order difference of the signal.
+
+        The default (0.97) matches the pre-emphasis filter used in the HTK
+        implementation of MFCCs [#]_.
+
+        .. [#] http://htk.eng.cam.ac.uk/
+
+    zi : number
+        Initial filter state. If inverting a previous preemphasis(), the same value should be used.
+
+        By default ``zi`` is initialized as
+        ``((2 - coef) * y[0] - y[1]) / (3 - coef)``. This
+        value corresponds to the transformation of the default initialization of ``zi`` in ``preemphasis()``,
+        ``2*x[0] - x[1]``.
+
+    Returns
+    -------
+    y_out : np.ndarray
+        pre-emphasized signal
+
+    Examples
+    --------
+    Apply a standard pre-emphasis filter and invert it with de-emphasis
+
+    >>> y, sr = librosa.load(librosa.ex('trumpet'))
+    >>> y_filt = librosa.effects.preemphasis(y)
+    >>> y_deemph = librosa.effects.deemphasis(y_filt)
+    >>> np.allclose(y, y_deemph)
+    True
+
+    See Also
+    --------
+    preemphasis
+    """
+    b = np.asarray([1.0, -coef], dtype=y.dtype)
+    a = np.asarray([1.0], dtype=y.dtype)
+
+    if zi is None:
+        if coef == 3.0:
+            raise NotImplementedError("Unable to do deemphasis if coefficient is 3.")
```

Comment:
i don't think we need to explicitly catch this exception, considering that we've already documented that the `coef` is expected to range between 0 and 1. Setting `coef` to `3.0` will naturally cause a `ZeroDivisionError`, which, i would argue, is exactly the right kind of error for this input.

@bmcfee feel free to object

Reply:
Removed in https://github.com/librosa/librosa/pull/1293/commits/c25e8c220b0e565473c16eeed82babed4ff1b1bd.

I was originally thinking that `coef=3` was a special case that should never come up, but could be handled in the future. Therefore, it was not yet implemented. But I took a second look at the math and I think the value of `zi` can't be deduced from the transformed signal alone if `coef=3`. Handling that case would require passing in more information, such as `x[0]` of the non-preemphasized signal. Anyway, as you said, it's reasonable to not handle that case because it's outside the range `0<coef<1`.